dropshot_endpoint/src/metadata.rs

```diff
@@ -239,6 +285,87 @@ impl ValidatedEndpointMetadata {
     }
 }
 
+#[derive(Debug)]
+pub(crate) enum VersionRange {
+    All,
+    From(semver::Version),
+    Until(semver::Version),
+    FromUntil(semver::Version, semver::Version),
+}
+
+fn parse_semver(v: &syn::LitStr) -> syn::Result<semver::Version> {
+    v.value()
+        .parse::<semver::Version>()
+        .map_err(|e| {
+            syn::Error::new_spanned(v, format!("expected semver: {}", e))
```

Comment:
what do Semver::parse() errors look like? Are those more useful than something like "{} is not of the form MAJOR.MINOR.PATCH"?

Reply:
I think the messages it produces are okay.  Here are a few examples:
https://github.com/oxidecomputer/dropshot/blob/e747169e4f2b834c0a03cd7411e38cf9fad64b38/dropshot_endpoint/src/endpoint.rs#L1070-L1085
https://github.com/oxidecomputer/dropshot/blob/e747169e4f2b834c0a03cd7411e38cf9fad64b38/dropshot_endpoint/src/endpoint.rs#L1094-L1109